app/assets/javascripts/comments.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+$(document).ready(function() {
```

Comment:
really all this JS looks strange to me. if it works - ok, but what I can suggest is to not use AJAX for such things in future on this project. 
1) because it's taking long time
2) because code is not ok, and normally it could be implemented differently (see "rails ajax remote true")

So, next time on other task - page could be refreshed/reloaded, which is ok